### PR TITLE
[CK_TILE] naive attn

### PR DESCRIPTION
**Pull request by carlushuang**
add a gpu implementation for attn(currently fwd). This is useful to do experimental works, while this is much faster than cpu reference impl

### CODE REVIEW

File: include/ck_tile/ref/naive_attention.hpp [hunk 1]
```diff
@@ -0,0 +1,472 @@
...
+    void* o_ptr;
+    void* context_len_ptr; // used when seqlen kv come from a pointer(each element is a number, not
+                           // cumsum)
+    void* page_table_ptr;  // [batch, num_blocks] seqlen_kv is in different block(paged attn)
+    float scale_s;
+    int hdim;
+    int hdim_v; // could be cross-attn, where V and Q/K hdim are different
+    int batch_q;
+    int batch_kv;
+    int batch_radio_kv; // batch_q / batch_kv
```
Comment (tenpercent) [inline include/ck_tile/ref/naive_attention.hpp]: nit: ratio

File: include/ck_tile/ref/naive_attention.hpp [hunk 2]
```diff
@@ -0,0 +1,472 @@
...
+            int page_offset  = get_phy_page_offset(i_s);
+            int64_t page_idx = get_phy_page_idx(i_s);
+            int64_t base_    = page_idx * h * s * d;
+            if constexpr(Layout == naive_attention_layout_enum::PHSD)
+                return static_cast<int64_t>(i_h * s * d + page_offset * d + i_d) + base_;
+            else if constexpr(Layout == naive_attention_layout_enum::PHSDX)
+            {
+                int x_r = page_offset % x;
+                int x_s = page_offset / x;
+                return static_cast<int64_t>(i_h * s * d + x_s * d * x + i_d * x + x_r) + base_;
```
Comment (tenpercent) [inline include/ck_tile/ref/naive_attention.hpp]: should it be `i_h * s * d * x`?

Comment (carlushuang) [reply include/ck_tile/ref/naive_attention.hpp]: the `s` here is for `page-size`, and `page-size` would be sub divided into `x`(vector size) and the remaining (name it `xs`). So we have `s = x * xs`, here this layout is still OK, cause we just ignore the subdivide, and multiply back process.
The key is, `s` always is `page-size`, no matter what layout

**Review by poyenc — APPROVED** [review]

**Review by rocking5566 — APPROVED** [review]
LGTM

**Pull request merged**